packages/flame_studio/lib/src/widgets/panels/hierarchy_view.dart

```diff
@@ -0,0 +1,240 @@
+import 'package:flame/components.dart';
+import 'package:flame_studio/src/core/component_tree.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+class HierarchyView extends ConsumerStatefulWidget {
+  const HierarchyView({super.key});
+
+  @override
+  HierarchyViewState createState() => HierarchyViewState();
+}
+
+class HierarchyViewState extends ConsumerState<ConsumerStatefulWidget> {
+  Component? selectedComponent;
+  Set<Component> expandedComponents = {};
+
+  @override
+  Widget build(BuildContext context) {
+    final componentTree = ref.watch(componentTreeProvider).root;
+    if (componentTree == null) {
+      return Container();
+    }
+
+    final listItems = <Widget>[];
+    _buildList(componentTree, 0, listItems);
+
+    return ListView(
+      padding: const EdgeInsets.all(8),
+      children: listItems,
+    );
+  }
+
+  void _buildList(ComponentTreeNode node, int depth, List<Widget> out) {
```

Comment:
I surprised there isn't already a "TreeComponent" on flutter that does this recursion for you

Reply:
The `ListView` already does most of the job, and conversion of a tree into a list is too ambiguous to be supported by the framework.